crates/loader/src/local/mod.rs

```diff
@@ -188,6 +189,45 @@ async fn core(
             let name = format!("{}@{}", bindle_id, parcel_sha);
             ModuleSource::Buffer(bytes, name)
         }
+        config::RawModuleSource::Url(us) => {
+            let url = reqwest::Url::parse(&us.url)
+                .with_context(|| format!("Invalid source URL {} in component {}", us.url, id))?;
+            if url.scheme() != "https" {
+                anyhow::bail!(
+                    "Invalid URL scheme {} in component {}: must be HTTPS",
+                    url.scheme(),
+                    id
+                );
+            }
+
+            let (digest_format, digest_text) = us.digest.split_once(':').ok_or_else(|| {
+                anyhow!(
+                    "Invalid digest string '{}' in component {}: must be sha256:...",
+                    us.digest,
+                    id
+                )
+            })?;
+            if digest_format != "sha256" {
```

Comment:
It's a bit messy to have this conditional format logic split between here and `verify_digest`. Perhaps simplify both to only support sha256 (YAGNI) or move all the logic to an e.g. `enum ContentDigest { Sha256(String) }` with a `TryFrom<String>`?